common/src/main/java/io/netty/util/concurrent/ScheduledTask.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2018 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.concurrent;
+
+/**
+ * Represents a task that is scheduled to expire some time in the future.
+ */
+public interface ScheduledTask {
```

Comment:
Can't we just use `Delayed` ?

https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Delayed.html

Reply:
We want to get the `deadline`, not the `delay`.

Reply:
got it

Reply:
@Scottmitch that said should we just make this part of the Abstract base class and mark as unstable ?

Reply:
What abstract base class? The issue is we need it to be accessible outside the package (`PromiseTask` and `ScheduledFutureTask` are package private). I considered putting a protected interface in `AbstractScheduledEventExecutor` and can do this to reduce the exposure of this API. I will push this change and we can discuss more if necessary.

On a related note ... the relationship between `ScheduledFutureTask` and `AbstractScheduledEventExecutor` is very tightly coupled. I would like to tease these two apart in general as there a assumptions needed about using the "correct" `nanoTime` that may be made less "leaky" for the next major release (I added something to the doc).